### PR TITLE
New dashboard initial and empty state placeholders

### PR DESCRIPTION
**Pull request by csubira**
https://github.com/CartoDB/cartodb/issues/14534

### CODE REVIEW

File: lib/assets/javascripts/new-dashboard/pages/Data.vue [hunk 1]
```diff
@@ -71,6 +71,11 @@
       </ul>
 
       <div class="grid-cell grid-cell--col12">
+        <EmptyState
```
Comment (jesusbotella) [inline lib/assets/javascripts/new-dashboard/pages/Data.vue]: Given that we are using the same component to render both states, why don't we change the translation to a computed property and combine both `v-if`?

It should be something like:
```html
<EmptyState
  :text="emptyStateText"
  v-if="emptyState || (initialState && hasSharedDatasets)">
    ...
</EmptyState>
```
* Use the name that you think it's better for the empty state text variable :)

Let me know if you think this approach is worse than what we have now :)

File: lib/assets/javascripts/new-dashboard/pages/Maps.vue [hunk 2]
```diff
@@ -60,6 +60,12 @@
         </li>
       </ul>
 
+      <EmptyState
```
Comment (jesusbotella) [inline lib/assets/javascripts/new-dashboard/pages/Maps.vue]: Same here!

File: lib/assets/javascripts/new-dashboard/pages/Data.vue (hunk 1 above)
Comment (csubira) [reply lib/assets/javascripts/new-dashboard/pages/Data.vue]: Yes! That's better that what we have at the moment. We can remove the duplicated component and it's still understandable! ;)

File: lib/assets/javascripts/new-dashboard/pages/Maps.vue [hunk 3]
```diff
@@ -154,6 +148,9 @@ export default {
     emptyState () {
       return !this.isFetchingMaps && !this.numResults && (!this.hasFilterApplied('mine') || this.totalUserEntries > 0);
     },
+    emptyStateText () {
+      return (this.initialState && this.hasSharedMaps) ? this.$t('MapsPage.emptyCase.onlyShared', { path: this.$router.options.base }) : this.$t('MapsPage.emptyCase.default', { path: this.$router.options.base });
```
Comment (jesusbotella) [inline lib/assets/javascripts/new-dashboard/pages/Maps.vue]: We can use [`resolve`](https://router.vuejs.org/api/#router-resolve) method to get the full URL based on the route name, and then pass it on to the translation.

**Pull request merged**